### ISSUE TITLE
[BUG] Automatic fan mode doesn't work

### ISSUE THREAD

**Issue by ClydeDroid** [rnd-ash/razer-laptop-control#21]
**Describe the bug**
When I put the fan in automatic mode, it turns off and never comes on, even when stress testing the CPU. However, I'm still able to control the fan speed manually in the range from 3500 to 5300.

**To Reproduce**
1. `sudo ./rgb_test --fan auto `
2. `stress --cpu 6`
3. Fan does not come on

**Expected behavior**
I would expect that when in automatic mode, the fan will respond to the CPU temperature and spin faster as the CPU gets hotter.

**Optional but useful, packet capture from synapse**
I don't know how to get this - would I need to run Windows in a VM from Linux? If so I'm willing to give it a shot if it helps with this project!

**Other details that might be helpful:**

Blade model: Mid 2019 Advanced Mercury White Blade 15 (Intel i7-9750H, RTX 2070)
OS: Arch Linux
Kernel: 5.4.8

Also I just wanted to give a shout-out to  @UKSFM99, this is really great work and it's awesome that you've chosen to share it with the community. I'm a software engineer and would love to help out however I can, I don't have much experience with C but am interested in learning!

**Comment by UKSFM99** [comment]
what laptop do you have? automatic firmware is self managed by the laptop so do the fans work normally without the driver?

**Comment by ClydeDroid** [comment]
This is the laptop I have: https://www.razer.com/gaming-laptops/razer-blade/shop#rtx2070--fhd-240hz-9thgen--512gb-mercury

I have a dual boot setup - the fans work in Windows, but not in Linux without the driver.

**Comment by UKSFM99** [comment]
Wait so the fans dont work AT ALL Under Linux!?

**Comment by ClydeDroid** [comment]
My fans do not work natively when booting into Linux, no. If they did, why would I need this driver? I was under the impression that the Linux kernel doesn't natively support Razer hardware including the fans and RGB keyboard, and that the point of this project was to implement that support? Am I incorrect in that assumption?

My fans do work when using your driver to set the fan speed explicitly, just not in automatic mode.

**Comment by rnd-ash** [comment]
This is really odd and will require extra work...

**Comment by nuudlz** [comment]
Hey ClydeDroid,
so your fans arent running at all and when you turn stress on your system will end up I a kernel panic to prevent damage (trip points being reached)?

Which distro are you running and could you post sensors about the temps and trip points?

Do you get any fan responses if you just switch the power modes from "balanced" to "gaming" with `sudo ./rgb_test --power gaming` ?

Be careful tho - burning your system isnt helpful at all.

**Comment by rnd-ash** [comment]
yeah ClydeDroid. If the fans simply don't work under Linux and the system shuts down when running stress without my driver installed, tell me and I shall try and find some way of raising this with Razer

**Comment by ClydeDroid** [comment]
Okay, so after some more testing, it turns out that the fans do work, but they just turn on at a much higher CPU temp than I was expecting.

I ran `stress --cpu 50` to peg my CPU at 100% and then opened up a video call at the same time, and the fans turned on after the CPU hit 51C. But even then, the fans were pretty quiet, much quieter than when I manually turn them up to 3500.

I guess I just assumed the fans didn't work because I was never working this thing hard enough in Linux, since I game in Windows only. :sweat_smile: 

Anyways, thanks @rnd-ash and @nuudlz for helping me look into this - and sorry if I wasted any time!
